web/src/main/java/com/synopsys/integration/alert/web/api/job/JobConfigActions.java

```diff
@@ -230,18 +231,17 @@ public JobConfigActions(AuthorizationManager authorizationManager, DescriptorAcc
         }
     }
 
-    private ValidationResponseModel validateJobNameUnique(@Nullable UUID currentJobId, JobFieldModel jobFieldModel) {
-        ValidationResponseModel responseModel = ValidationResponseModel.success("Valid name");
+    private Optional<AlertFieldStatus> validateJobNameUnique(@Nullable UUID currentJobId, JobFieldModel jobFieldModel) {
         for (FieldModel fieldModel : jobFieldModel.getFieldModels()) {
-            responseModel = validateJobNameUnique(currentJobId, fieldModel);
-            if (responseModel.hasErrors()) {
-                return responseModel;
+            Optional<AlertFieldStatus> fieldStatus = validateJobNameUnique(currentJobId, fieldModel);
```

Comment:
Could you use a stream here to remove the for loop?  You would have to change the method signature of validateJobNameUnique to take a FieldValueModel.

```
Optional<AlertFieldStatus> fieldStatus = jobFieldModel.getFieldModels().stream()
     .filter(fieldModel ->                                            
                     fieldModel.getFieldValueModel(ChannelDistributionUIConfig.KEY_NAME).isPresent())
     .map(fieldModel.getFieldValueModel(ChannelDistributionUIConfig.KEY_NAME))
     .map(fieldValueModel -> validateJobNameUnique(currentJobId, fieldValueModel)
     .orElse(Optional.empty());
```